[2025-03-16T15:28:39.173+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-16T15:28:39.280+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: skripsi_perbaikan.predict_and_store_result scheduled__2025-03-16T15:26:00+00:00 [queued]>
[2025-03-16T15:28:39.368+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: skripsi_perbaikan.predict_and_store_result scheduled__2025-03-16T15:26:00+00:00 [queued]>
[2025-03-16T15:28:39.372+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 4
[2025-03-16T15:28:39.586+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): predict_and_store_result> on 2025-03-16 15:26:00+00:00
[2025-03-16T15:28:39.727+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=77588) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-03-16T15:28:39.737+0000] {standard_task_runner.py:72} INFO - Started process 77704 to run task
[2025-03-16T15:28:39.765+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'skripsi_perbaikan', 'predict_and_store_result', 'scheduled__2025-03-16T15:26:00+00:00', '--job-id', '606', '--raw', '--subdir', 'DAGS_FOLDER/tugas_akhir.py', '--cfg-path', '/tmp/tmpvxghpsn4']
[2025-03-16T15:28:39.775+0000] {standard_task_runner.py:105} INFO - Job 606: Subtask predict_and_store_result
[2025-03-16T15:28:40.091+0000] {task_command.py:467} INFO - Running <TaskInstance: skripsi_perbaikan.predict_and_store_result scheduled__2025-03-16T15:26:00+00:00 [running]> on host 438c3ba1e399
[2025-03-16T15:28:40.892+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Dion Rikki' AIRFLOW_CTX_DAG_ID='skripsi_perbaikan' AIRFLOW_CTX_TASK_ID='predict_and_store_result' AIRFLOW_CTX_EXECUTION_DATE='2025-03-16T15:26:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-16T15:26:00+00:00'
[2025-03-16T15:28:40.935+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-03-16T15:28:40.943+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-03-16T15:28:40.952+0000] {logging_mixin.py:190} INFO - Current task name:predict_and_store_result state:running start_date:2025-03-16 15:28:39.286661+00:00
[2025-03-16T15:28:40.957+0000] {logging_mixin.py:190} INFO - Dag name:skripsi_perbaikan and current dag run status:running
[2025-03-16T15:28:40.963+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-03-16T15:28:40.978+0000] {logging_mixin.py:190} INFO - üîç [2] Memproses data dan melakukan prediksi...
[2025-03-16T15:28:41.142+0000] {logging_mixin.py:190} INFO - ‚úÖ Data siap diproses: 1 baris
   id  Temperature  Humidity  ...  nitrogen phosphorous potassium
0  80         24.8      85.0  ...       0.0         0.0       0.0

[1 rows x 9 columns]
[2025-03-16T15:28:41.146+0000] {logging_mixin.py:190} INFO - üìù Menyesuaikan nama kolom dengan model training...
[2025-03-16T15:28:41.149+0000] {logging_mixin.py:190} INFO - üîß Melakukan encoding kategori Soil_Type dan Crop_Type...
[2025-03-16T15:28:41.171+0000] {logging_mixin.py:190} INFO - üß† Menyiapkan fitur untuk prediksi...
[2025-03-16T15:28:41.186+0000] {logging_mixin.py:190} INFO - ü§ñ Menjalankan model prediksi...
[2025-03-16T15:28:41.299+0000] {logging_mixin.py:190} INFO - ‚úÖ Prediksi selesai:
   id predicted_fertilizer
0  80             14-14-14
[2025-03-16T15:28:41.306+0000] {logging_mixin.py:190} INFO - üíæ Menyimpan hasil ke tabel fertilizer_predictions...
[2025-03-16T15:28:41.500+0000] {logging_mixin.py:190} INFO - ‚ùå Error saat menyimpan ke database: (pymysql.err.IntegrityError) (1062, "Duplicate entry '80' for key 'fertilizer_predictions.sensor_data_id'")
[SQL: INSERT INTO fertilizer_predictions (sensor_data_id, predicted_fertilizer, created_at) VALUES (%(sensor_data_id)s, %(predicted_fertilizer)s, %(created_at)s)]
[parameters: {'sensor_data_id': 80, 'predicted_fertilizer': '14-14-14', 'created_at': datetime.datetime(2025, 3, 16, 15, 28, 41, 327290)}]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2025-03-16T15:28:41.503+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-03-16T15:28:41.621+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-03-16T15:28:41.636+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=skripsi_perbaikan, task_id=predict_and_store_result, run_id=scheduled__2025-03-16T15:26:00+00:00, execution_date=20250316T152600, start_date=20250316T152839, end_date=20250316T152841
[2025-03-16T15:28:41.732+0000] {logging_mixin.py:190} INFO - Task instance in success state
[2025-03-16T15:28:41.742+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: running
[2025-03-16T15:28:41.753+0000] {logging_mixin.py:190} INFO - Dag name:skripsi_perbaikan queued_at:2025-03-16 15:28:01.694711+00:00
[2025-03-16T15:28:41.762+0000] {logging_mixin.py:190} INFO - Task hostname:438c3ba1e399 operator:PythonOperator
[2025-03-16T15:28:41.893+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-03-16T15:28:42.026+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-16T15:28:42.036+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
